### PR TITLE
fix(MeshConfig): adding rbac for deleting meshConfig

### PR DESCRIPTION
**Pull request by snehachhabria**
**Description**:

The PR adds rbac to delete meshconfig resources in osm, while using the helm cleanup hooks. This was causing issues in the CI

Signed-off-by: Sneha Chhabria snchh@microsoft.com

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [X] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [X] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? `no`
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change? `no`

### CODE REVIEW

File: charts/osm/templates/osm-rbac.yaml [hunk 1]
```diff
@@ -90,7 +90,7 @@ rules:
     verbs: ["list"]
   - apiGroups: ["config.openservicemesh.io"]
     resources: ["meshconfigs"]
-    verbs: ["get", "list", "watch", "create", "update", "patch"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```
Comment (shashankram) [inline charts/osm/templates/osm-rbac.yaml:93]: Duplicate present in L104

Comment (snehachhabria) [reply charts/osm/templates/osm-rbac.yaml:93]: removed duplicate

**Review by nojnhuh — APPROVED** [review]

File: charts/osm/templates/cleanup-hook.yaml [hunk 2]
```diff
@@ -26,3 +26,4 @@ spec:
             - -c
             - >
              kubectl delete --ignore-not-found meshconfig -n '{{ include "osm.namespace" . }}' osm-mesh-config;
+             kubectl delete --ignore-not-found meshconfig -n '{{ include "osm.namespace" . }}' preset-mesh-config;
```
Comment (shashankram) [inline charts/osm/templates/cleanup-hook.yaml:29]: Why is this required? Shouldn't this be automatically deleted by Helm?

Comment (snehachhabria) [reply charts/osm/templates/cleanup-hook.yaml:29]: the configmap also previously was installed by helm but this cleanup hook handled it, hence I added the preset-mesh-config

cc @nojnhuh

Comment (nojnhuh) [reply charts/osm/templates/cleanup-hook.yaml:29]: We added the configmap deletion here in anticipation of removing it from the chart. Since the preset-mesh-config is part of the chart, we don't need to delete it here.

Comment (shashankram) [reply charts/osm/templates/cleanup-hook.yaml:29]: Hmm, unsure why this would be necessary. @nojnhuh was there a reason to explicitly remove this?

**Review by shashankram — APPROVED** [review]

File: charts/osm/templates/cleanup-hook.yaml (hunk 2 above)
Comment (snehachhabria) [reply charts/osm/templates/cleanup-hook.yaml]: thanks for the clarification @nojnhuh , I have removed the preset-mesh-config
